src/socket.c

```diff
@@ -284,8 +284,15 @@ gint socket_init(gint argc, gchar **argv)
 	GdkDisplay *display = gdk_display_get_default();
 	gchar *p;
 
+	/* On OS X with quartz backend gdk_display_get_name() returns hostname
+	 * using [NSHost currentHost] (it could return more or less whatever string
+	 * as display name is a X11 specific thing). This call can lead to network
+	 * query and block for several seconds so better skip it. */
+#ifndef GDK_WINDOWING_QUARTZ
```

Comment:
maybe under Quartz it doesn't matter, but shouldn't this be guarded by `if (GDK_IS_QUARTZ_DISPLAY (display))` to support a build combining X11 and Quartz?

```C
#ifdef GDK_WINDOWING_QUARTZ
	if (display != NULL && GDK_IS_QUARTZ_DISPLAY (display))
		; /* nothing */
	else
#endif
	if (display != NULL)
		...
```
Or a prettier variation around this.